sda-commons-starter-kafka/src/main/java/org/sdase/commons/spring/boot/kafka/KafkaHealthIndicator.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2022- SDA SE Open Industry Solutions (https://www.sda.se)
+ *
+ * Use of this source code is governed by an MIT-style
+ * license that can be found in the LICENSE file or at
+ * https://opensource.org/licenses/MIT.
+ */
+package org.sdase.commons.spring.boot.kafka;
+
+import java.time.Duration;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import org.apache.kafka.clients.admin.AdminClient;
+import org.apache.kafka.clients.admin.DescribeClusterOptions;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.actuate.autoconfigure.health.ConditionalOnEnabledHealthIndicator;
+import org.springframework.boot.actuate.health.AbstractHealthIndicator;
+import org.springframework.boot.actuate.health.Health;
+import org.springframework.kafka.core.KafkaAdmin;
+import org.springframework.stereotype.Component;
+
+@Component("kafka")
+@ConditionalOnEnabledHealthIndicator("kafka")
+public class KafkaHealthIndicator extends AbstractHealthIndicator {
+
+  private static final Logger LOG = LoggerFactory.getLogger(KafkaHealthIndicator.class);
+
+  private final AdminClient kafkaAdminClient;
+  private final Duration kafkaCommandTimeout;
+
+  public KafkaHealthIndicator(
+      @Value("${management.health.kafka.timeout:4s}") Duration kafkaCommandTimeout,
+      KafkaAdmin kafkaAdmin) {
+    super("Kafka health check failed");
+
+    this.kafkaCommandTimeout = kafkaCommandTimeout;
+    this.kafkaAdminClient = AdminClient.create(kafkaAdmin.getConfigurationProperties());
+
+    LOG.info("Kafka health check is initialized with timeout duration {}", kafkaCommandTimeout);
+  }
+
+  @Override
+  protected void doHealthCheck(Health.Builder builder) throws Exception {
+    var clusterInfo =
+        kafkaAdminClient.describeCluster(
+            new DescribeClusterOptions()
+                .timeoutMs((int) TimeUnit.MILLISECONDS.convert(kafkaCommandTimeout)));
```

Comment:
Have you double-checked that this will not be a (missing) privilege issue? Are users that only have access to a specific topic able to read that information?

Reply:
I tested it via:
- Integration test using `embeddedKafka`
- Spinning up local-infra and connecting to kafka using locally running service.

I didn't encounter any issue by using it, but I didn't test only for user that has access to certain topic as I am not sure how to configure it.

Reply:
The Dropwizard Library uses [list topics](https://github.com/SDA-SE/sda-dropwizard-commons/blob/master/sda-commons-server-kafka/src/main/java/org/sdase/commons/server/kafka/health/KafkaHealthCheck.java#L29) for health. I can't see any disabling of the health check and it seems to work with restrictive ACLs.

Reply:
@JoergSiebahn, what did you mean by 
> I can't see any disabling of the health check
Disable health feature is implemented via [@ConditionalOnEnabledHealthIndicator](https://docs.spring.io/spring-boot/docs/current/api/org/springframework/boot/actuate/autoconfigure/health/ConditionalOnEnabledHealthIndicator.html) that gives that out-of-the-box

Reply:
That was about the practical use of the Dropwizard library. As the service using Dropwizard with Kafka in an environment with only topic related ACLs granted for the user is running and healthy, I assume privileges for list topics are granted when the user has access to a specific topic. The assumption and the prove by a working service could be wrong, if the health check is be disabled. As I can't find anything that disables the health check, I still assume, that list topics is a good call for a health check in services that produce or consume Kafka topics.

Reply:
Alright, let's use the same idea as for dropwizard services. Logic is adjusted respectively.